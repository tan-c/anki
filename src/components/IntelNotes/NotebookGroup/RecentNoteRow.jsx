import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Map } from 'immutable';

import { UiActions } from 'utility-redux/ui';
import { currentUserRecentNoteSelector } from 'utility-redux/user';

export class RecentNoteRow extends React.Component {
  // constructor(props, context) {
  //  super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const {
      recentNote,
    } = this.props;

    return (
      <div
        data-role="recentnote-row"
        className="border-bottom-white flex-container-row typical-setup overflow-hidden"
      >
        <i className="fa fa-fw fa-eye wdith-20" />
        {recentNote.size > 0
        && <div className="flex-1" role="menuitem" tabIndex="-1" />
        }

        {recentNote.size === 0
          && <div className="flex-1"> No Recent</div>
        }
      </div>
    );
  }
}

RecentNoteRow.defaultProps = {
  recentNote: Map(),
};

RecentNoteRow.propTypes = {
  recentNote: PropTypes.object,
};

function mapStateToProps(state, ownProps) {
  return {
    recentNote: currentUserRecentNoteSelector(state),
  };
}

function mapDispatchToProps(dispatch) {
  return {
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(RecentNoteRow);
