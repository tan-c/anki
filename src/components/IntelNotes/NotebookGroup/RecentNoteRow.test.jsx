import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

import { Map } from 'immutable';

import { RecentNoteRow } from './RecentNoteRow';

function setup() {
  const props = {
    recentNote: Map(),

    UiActions: {},
  };

  return shallow(<RecentNoteRow {...props} />);
}

describe('RecentNoteRow Unit Tests', () => {
  describe('', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      // expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("div[data-role='recentnote-row']")).toHaveLength(1);
    });
  });
});
