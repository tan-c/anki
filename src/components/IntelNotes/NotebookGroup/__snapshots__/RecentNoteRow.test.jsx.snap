// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentNoteRow Unit Tests  renders correctly 1`] = `
<div
  className="border-bottom-white flex-container-row typical-setup overflow-hidden"
  data-role="recentnote-row"
>
  <i
    className="fa fa-fw fa-eye wdith-20"
  />
  <div
    className="flex-1"
  >
     No Recent
  </div>
</div>
`;
